rmgpy/data/solvation.py

```diff
@@ -1848,25 +1848,56 @@ def get_Kfactor(self, solute_data, solvent_data, T):
         else:
             raise DatabaseError("K-factor calculation or temperature-dependent solvation free energy calculation "
-                                "is not available for the solvent whose `name_in_coolprop` is None")
+                                f"is not available for the given solvent name: {solvent_name}")
         return Kfactor
 
-    def get_T_dep_solvation_energy(self, solute_data, solvent_data, T):
+    def get_T_dep_solvation_energy_from_LSER_298(self, solute_data, solvent_data, T):
         """
-        Given solute_data, solvent_data, and temperature, calculates the Gibbs free energy of
-        solvation at T if the solvent's name_in_coolprop is not None.
+        Given solute_data, solvent_data, and temperature, calculates the Gibbs free energy of solvation and K-factor
+        at T if the solvent's name_in_coolprop is not None and the solvent has all solvent parameters.
+        Otherwise, it raises DatabaseError.
         """
-        Kfactor = self.get_Kfactor(solute_data, solvent_data, T)
-        rho_g = PropsSI('Dmolar', 'T', T, 'Q', 1, solvent_data.name_in_coolprop) # saturated gas phase density of the solvent, in mol/m^3
-        rho_l = PropsSI('Dmolar', 'T', T, 'Q', 0, solvent_data.name_in_coolprop) # saturated liquid phase density of the solvent, in mol/m^3
+
+        solvent_name = solvent_data.name_in_coolprop
+        # check whether all solvent parameters exist
+        solvent_param_list = [solvent_data.s_g, solvent_data.b_g, solvent_data.e_g,
+                              solvent_data.l_g, solvent_data.a_g, solvent_data.c_g,
+                              solvent_data.s_h, solvent_data.b_h, solvent_data.e_h,
+                              solvent_data.l_h, solvent_data.a_h, solvent_data.c_h,
+                              ]
+        param_found_list = [param is not None for param in solvent_param_list]
+
+        if solvent_name is not None and all(param_found_list):
+            correction = self.get_solvation_correction(solute_data, solvent_data)
+            delG298 = correction.gibbs  # in J/mol
+            delH298 = correction.enthalpy  # in J/mol
+            delS298 = correction.entropy  # in J/mol/K
+        else:
+            raise DatabaseError("K-factor parameter calculation is not available for the solvent "
+                                "whose `name_in_coolprop` is None or that is missing the Abraham and Mintz "
+                                "solvent parameters.")
+
+        Kfactor = self.get_Kfactor(delG298, delH298, delS298, solvent_name, T)
+        rho_g = PropsSI('Dmolar', 'T', T, 'Q', 1, solvent_name) # saturated gas phase density of the solvent, in mol/m^3
+        rho_l = PropsSI('Dmolar', 'T', T, 'Q', 0, solvent_name) # saturated liquid phase density of the solvent, in mol/m^3
         delG = constants.R * T * math.log(Kfactor * rho_g / (rho_l))  # in J/mol
-        return delG
+        return delG, Kfactor
```

Comment:
This code is repeated below. You might be able to make a method out of just this small portion so that it only has to be maintained in one place